### ISSUE TITLE
Errors in logstash since upgrade to 2.2

### ISSUE THREAD

**Issue by chad-upton** [logstash-plugins/logstash-input-beats#46]
After upgrading logstash to the newest version that came out today, I'm seeing these messages in the logs constantly using the beats input plugin.

```
{:timestamp=>"2016-02-06T06:21:15.454000+0000", :message=>"Beats Input: Remote connection closed", :peer=>"10.0.5.242:11572", :exception=>#<Lumberjack::Beats::Connection::ConnectionClosed: Lumberjack::Beats::Connection::ConnectionClosed wrapping: EOFError, End of file reached>, :level=>:warn}
{:timestamp=>"2016-02-06T06:21:19.290000+0000", :message=>"Beats Input: Remote connection closed", :peer=>"10.0.1.169:28246", :exception=>#<Lumberjack::Beats::Connection::ConnectionClosed: Lumberjack::Beats::Connection::ConnectionClosed wrapping: EOFError, End of file reached>, :level=>:warn}
```

**Comment by plumpNation** [comment]
I am also seeing the same issue.

However, it was working earlier. I am removing the registry and elasicsearch indices and pushing all the logs in again. Worked several times without issue, then stopped.

Have reverted all changes to the working setup, still the issue persists.

**Comment by ph** [comment]
@chad-upton In the latest release of the plugin I have added more logging, in the previous version we have similar errors but were not reporting them.

I you look at your filebeat logs do you see any errors related in that time frame?

@plumpNation In your case it is normal you are closing the remote connection which trigger the EOF on the logstash.

**Comment by ph** [comment]
@plumpNation Thanks we are trying our best to help out :) You should look at our discuss forum.  https://discuss.elastic.co

**Comment by chad-upton** [comment]
@ph It's a constant stream of that error as soon as logstash is started.

My logstash collectors are behind an AWS ELB, nothing showing up in the filebeat logs.

Appreciate the help!

**Comment by ph** [comment]
@chad-upton Can you share both your logstash and filebeat configuration?

**Comment by chad-upton** [comment]
Filebeat.yml
```
############################# Filebeat ######################################
filebeat:
  prospectors:
    -
      # Linux system logs
      paths:
        - /var/log/messages
        - /var/log/secure
        - /var/log/cloud-init-output.log
        - /var/log/yum.log
      encoding: utf-8
      input_type: system
      fields:
        json: false
      fields_under_root: true
    -
      # Puma Logs
      paths:
        - /var/log/puma/puma.log
      encoding: utf-8
      input_type: puma
      fields:
        json: false
      fields_under_root: true
    -
      # Rails worker
      paths:
        - /var/log/worker/*.log
        - /var/log/worker/*.err
      encoding: utf-8
      input_type: railsworker
      fields:
        json: false
      fields_under_root: true
    -
      # Nginx logs
      paths:
        - /var/log/nginx/*.log
      encoding: utf-8
      input_type: nginx
      fields:
        json: false
      fields_under_root: true
    -
      # Ossec
      paths:
        - /var/ossec/logs/alerts/alerts.log
      encoding: utf-8
      input_type: ossec
      fields:
        json: false
      fields_under_root: true
    -
      # Elasticbeanstalk
      paths:
        - /var/log/eb-activity.log
        - /var/log/eb-version-deployment.log
      encoding: utf-8
      input_type: elasticbeanstalk
      fields:
        json: false
      fields_under_root: true

  registry_file: /var/lib/filebeat/registry
output:
  logstash:
    hosts: ["logs.example.com:5044"]
    worker: 1

shipper:
logging:
  files:
    rotateeverybytes: 10485760 # = 10MB
```

I have multiple filters but only a single input from filebeat.

logstash.conf.d/
```
# All client side logs come in via this method

input {
  beats {
    port => 5044
  }
}
```

**Comment by ph** [comment]
Not sure its its not the ELB the problem in this case, filebeat can already load balance the requests between multiples host. 

You said it was working before? Which version of logstash where you running and which version of the beat plugin? Did you always used a ELB between filebeat and Logstash?

**Comment by chad-upton** [comment]
Oh it's working fine, it's just spamming the hell out of the logstash log file.  The only thing that has changed is I went to the new logstash version, was one back before it.  2.1.9 I think?

**Comment by ph** [comment]
Ok, I'll try to reproduce the same scenario that you have, I haven't see the EOF really spammy on my end :)

**Comment by personias** [comment]
I'm getting the same annoyance with logstash 2.2.0.1 & the logstash-input-beats plugin sitting behind an AWS ELB.  My ELB is running an 'every 30 second healthcheck'  (Ping of target host via TCP:5044).  When the healthcheck runs, I get that warning for each of the hosts in each of the ELBs (There is 1 per availability zone...so if you have a 2 AZ setup; you'll have 2 warnings).  The result of the healthcheck is a sucess, so again, it's all good...just ugly.  Like chad-upton, it appears to be otherwise working (I upgraded due to some connectivity issues with an earlier version of Logstash).  It's just spammy.

**Comment by ph** [comment]
@personias I will turn it down a bit, I haven't considered the health checks to cause this problem.

**Comment by personias** [comment]
@ph -- many thanks!

**Comment by shaunrampersad** [comment]
I'm seeing the same issue. My Logstash nodes are fronted by haproxy. Filebeat then talks to haproxy. I'm seeing the constant message 
'
{:timestamp=>"2016-02-19T16:07:12.459000+0200", :message=>"Beats Input: Remote connection closed", :peer=>"172.x.x.x:58134", :exception=>#<Lumberjack::Beats::Connection::ConnectionClosed: Lumberjack::Beats::Connection::ConnectionClosed wrapping: EOFError, End of file reached>, :level=>:warn}
{:timestamp=>"2016-02-19T16:07:18.496000+0200", :message=>"Beats Input: Remote connection closed", :peer=>"172.x.x.x:58180", :exception=>#<Lumberjack::Beats::Connection::ConnectionClosed: Lumberjack::Beats::Connection::ConnectionClosed wrapping: EOFError, End of file reached>, :level=>:warn}
{:timestamp=>"2016-02-19T16:08:08.085000+0200", :message=>"Beats Input: Remote connection closed", :peer=>"172.x.x.x:58741", :exception=>#<Lumberjack::Beats::Connection::ConnectionClosed: Lumberjack::Beats::Connection::ConnectionClosed wrapping: EOFError, End of file reached>, :level=>:warn}
'

I'm using Logstash 2.2.1

**Comment by ph** [comment]
I just pushed 2.1.4 which fix this issue!

**Comment by highbass** [comment]
this is still visible on version 2.2.2 ... are teh cahnges supposed to be merged up to 2.2.x?

**Comment by highbass** [comment]
i have version 2.2.2 ... looks like the changes did not make it into 2.2.x ?


{:timestamp=>"2016-07-19T00:58:38.710000-0400", :message=>"Beats Input: Remote connection closed", :peer=>"X.X.X.X:50655", :exception=>#<Lumberjack::Beats::Connection::ConnectionClosed: Lumberjack::Beats::Connection::ConnectionClosed wrapping: EOFError, End of file reached>, :level=>:warn}


seeing a flood of these

**Comment by ph** [comment]
@highbass I suspect that the 2.2.2 version that you mention is actually the logstash version?
Plugins and logstash have independent version, 2.1.4 was the plugin version, I dont remember what version is shipped with 2.2.2 but you can retrieve it by doing this command.

`bin/plugin list --verbose beats`

If the version is not 2.1.4 or higher you can update it with the following command:


`bin/plugin update logstash-input-beats`

This should fixes your issues.
